projects/wave-core-new/src/lib/project/project.service.ts

```diff
@@ -478,185 +256,84 @@ export class ProjectService {
-    //                 this.createRasterLayerDataSubscription(
-    //                     layer as RasterLayer<AbstractRasterSymbology>,
-    //                     (this.layerData$.get(layer) as Observer<RasterData>),
-    //                     this.layerDataState$.get(layer)
-    //                 )
-    //             );
-    //             this.layerSymbologyDataSubscriptions.set(layer,
-    //                 this.createRasterLayerSymbologyDataSubscription(
-    //                     layer as RasterLayer<AbstractRasterSymbology>,
-    //                     this.layerSymbologyData$.get(layer) as Observer<DeprecatedMappingColorizerDoNotUse>,
-    //                     this.layerSymbologyDataState$.get(layer)));
-    //             break;
-    //         }
-    //         case 'vector': {
-    //             this.layerDataSubscriptions.set(layer,
-    //                 this.createVectorLayerDataSubscription(
-    //                     layer as VectorLayer<AbstractVectorSymbology>,
-    //                     (this.layerData$.get(layer) as Observer<VectorData>),
-    //                     this.layerDataState$.get(layer)
-    //                 )
-    //             );
-    //             break;
-    //         }
-    //
-    //     }
-    //
-    //     this.layerProvenanceDataSubscriptions.set(layer,
-    //         this.createLayerProvenanceSubscription(
-    //             layer,
-    //             this.layerProvenanceData$.get(layer),
-    //             this.layerProvenanceDataState$.get(layer)
-    //         )
-    //     );
-    // }
+    /**
+     * Reload the data of a layer.
+     */
+    reloadLayerData(layer: Layer) {
+        this.layerData$.get(layer.id).next(undefined); // send empty data
+
+        if (this.layerDataSubscriptions.has(layer.id)) {
+            this.layerDataSubscriptions.get(layer.id).unsubscribe();
+            this.layerDataSubscriptions.delete(layer.id);
+        }
+
+        switch (layer.layerType) {
+            case 'raster': {
+                this.layerDataSubscriptions.set(layer.id,
+                    this.createRasterLayerDataSubscription(
+                        layer as RasterLayer,
+                        (this.layerData$.get(layer.id) as Observer<RasterData>),
+                        this.layerDataState$.get(layer.id)
+                    )
+                );
+                break;
+            }
+            case 'vector': {
+                this.layerDataSubscriptions.set(layer.id,
+                    this.createVectorLayerDataSubscription(
+                        layer as VectorLayer,
+                        (this.layerData$.get(layer.id) as Observer<VectorData>),
+                        this.layerDataState$.get(layer.id)
+                    )
+                );
+                break;
+            }
+
+        }
+    }
+
+    /**
+     * Reload everything for the layer manually (e.g. on error).
+     */
+    reloadLayer(layer: Layer) {
```

Comment:
this is the same code as the method above?

Reply:
Yes, so I call it now.

904f41c 

There will be most likely more stuff that is necessary to load a layer (previously e.g. provenance)